app/services/task-assignment.js

```diff
@@ -29,16 +29,25 @@ export default Service.extend({
 
   _create(user, task) {
     let store = get(this, 'store');
-    return store.createRecord('user-task', { user, task })
-                .save();
+    let userTask = store.createRecord('user-task', { user, task });
+    return userTask.save().catch(() => {
+      userTask.deleteRecord();
```

Comment:
Nitpicking, `deleteRecord` is not an api call. `destroyRecord` is. 

However, you are right. `unloadRecord` is the correct choice here.